### PR TITLE
[FIX] web_responsive: Hide footer instead of remove

### PR DESCRIPTION
**Pull request by lasley**
- Add style to hide oe_footer instead of remove to not break `support_branding`
- Add note in ReadMe explaining override of `support_branding`
- Add `oe_main_menu_navbar` id to navbar

This is the v9 fix for `support_branding` in #445

### CODE REVIEW

**Comment by pedrobaeza** [comment]
We should take the occasion to hide all replaced elements instead of just removing them, so we maximize the compatibility with other possible modules.

One thing: what about putting the company logo a bit more smaller in the upper right corner of the drawer?

**Comment by lasley** [comment]
I originally hid everything, but was told by @Yajo to remove instead in the following comments:
- https://github.com/OCA/web/pull/396#r76193405
- https://github.com/OCA/web/pull/396#r76193550
- https://github.com/OCA/web/pull/396#r76193557
- https://github.com/OCA/web/pull/396#r76193727
- https://github.com/OCA/web/pull/396#r76193743

So I should revert all that? I agree with his logic TBH - there's a lot of JS happening & menus getting filled in that adds a lot of load to the page without replacing.

**Comment by pedrobaeza** [comment]
When there's JS logic involved, I agree on the optimization principle, but we don't know if other modules will mess up with these removals. Is it so heavy to let them?

**Comment by lasley** [comment]
Regarding the company logo - I can add in the title pane of the app drawer in v9, but it shrunk considerably in v10 and I don't think would be viable:

image

vs

image

I admittedly like the smaller UI of 10, so I hesitate to change. Is the missing logo something that will bug people you think?

**Comment by lasley** [comment]
It's pretty heavy. If I leave them, the entire menus are duplicated. The interesting thing about the way the menus are generated too is that every menu is added, but hidden. This entire section would be duplicated - 4.5k lines of code on my local:

image

**Comment by pedrobaeza** [comment]
There are 2 cons for not having logo:
- The company branding, that all company managers wants to have in their systems (well, everywhere)
- In multi-company environments, users look at the logo to see in which company they are.

**Comment by pedrobaeza** [comment]
OK, let's see what we get with the replacings. Don't touch that part for now.

**Comment by pedrobaeza** [comment]
Even smaller, in v10 the logo can be there as something testimonial (or include a system parameter to configure the height at request)

**Comment by lasley** [comment]
It shouldn't be too difficult to add the logo in the bottom center of the App Drawer. Will that work?

**Comment by pedrobaeza** [comment]
Well, if we put it with absolute positioning plus float feature, it would be enough, but it's useless if the logo is not shown when opening because you have to scroll down.

**Comment by lasley** [comment]
Agreed. I just looked at Enterprise for some possible inspiration, but it seems they put it at the bottom and required scroll.

I can increase the height of the App Drawer title bar in 10 to support it at the top then. I can't really think of a better spot for it, and there is space there. I am planning on adding a Search Bar there too, but that can go top middle. Or maybe the logo top middle and search top right.

I don't think a param will work because the height is controlled in CSS, so I wouldn't be able to grab the param (at least I don't think so, let me know if I'm missing some Odoo magic).

**Comment by pedrobaeza** [comment]
Isn't better to put the logo inside the top bar instead of overlapping the apps icons?

**Comment by lasley** [comment]
Yeah that was the intent, and what it looked like on my local - This definitely wasn't what I was going for 😆

**Comment by pedrobaeza** [comment]
hehe, we definitively need that docker that mimicks Runbot behavior on local :wink:

**Comment by lasley** [comment]
Seriously - my dev environment situation sucks at the moment. This is the first I've bothered with the other Odoo versions, which turns out to be a real pain. 

Something will be figured out, sooner rather than later - that's for sure. My poor juniors will be so 😕  with this workflow.

**Comment by lasley** [comment]
Rebased for Travis changes so my Runbot doesn't skip the builds anymore. The last commit should hopefully fix the logo issue too

**Comment by lasley** [comment]
Note to self, looks like junk on mobile 

image

**Comment by lasley** [comment]
Annnnnd now it doesn't! This is ready for review again I believe.

Note that this is still incompatible with the `support_branding` module as detailed in the ReadMe (build isn't broken, but `support_branding` assets are simply hidden). A glue module will need to be created in order for the `support_branding` stuff to be injected somewhere other than `web.secondary_menu`, which is the top bar in this interface.

**Comment by pedrobaeza** [comment]
Thanks for the efforts, Dave. I have tried and I feel that the logo is a bit small, leaving a big margin on top and bottom. Or is this to be the same for v10?

About support_branding, we can remove the word incompatible, because as said it doesn't fail, but one of the features isn't simply shown. But the other is still there: the error dialog includes the possibility of sending the traceback to a support mail.

**Comment by lasley** [comment]
Alright updated the ReadMe. Logo sizing is better now too

image

**Comment by lasley** [comment]
Dang now that I see the ruler guides against the "Apps" link, I realize that the Apps link is not center. That's going to bug me now that I know about it, so will fix

**Comment by pedrobaeza** [comment]
This looks much better, but I'm afraid there's still one thing to do: restore the "Edit company data" link. In v9, without it, you can't edit company data, unless you activate "Multi companies".

File: web_responsive/views/web.xml
Comment (yajo) [inline web_responsive/views/web.xml]: Better add class `hidden` if possible.

File: web_responsive/views/web.xml
Comment (yajo) [inline web_responsive/views/web.xml]: Can't you use `keep_query`?

File: web_responsive/views/web.xml
Comment (yajo) [inline web_responsive/views/web.xml]: You'll probably get dirty translations with this. Better use a div for fontawesome icons or add `t-translation="off"` please.

**Review by yajo — CHANGES_REQUESTED** [review]

File: web_responsive/views/web.xml
Comment (lasley) [reply web_responsive/views/web.xml]: Is there a way to append into the existing class? The goal here is to allow compatibility with addons that may target the div. At the moment it's `oe_footer`, but I don't know if other modules inject classes into that or other things.

Comment (yajo) [reply web_responsive/views/web.xml]: Then why not just add `.oe_footer {display:none}` to the Less file?

Comment (lasley) [reply web_responsive/views/web.xml]: Because I didn't think of it? 😆  👍

**Comment by lasley** [comment]
Alright we should be good. I added a bonus fix on the toggle button padding/border too 😄 

Note the padding had to be made a bit smaller due to [adding of `oe_main_menu_navbar` id](https://github.com/OCA/web/pull/446/commits/0ae026047a364f56c2f2326db9c82980ac3848e3#diff-b2e18ea72a902d3738259f602168f4d5R38) which triggered some new CSS. Looks more like the v10 branch now, which makes me happy.

**Comment by pedrobaeza** [comment]
Very good! Just one minor thing: edit icon shouldn't be visible when you're not admin, as you don't have rights to edit company data.

**Comment by lasley** [comment]
Fixed - brought in the [v10 style](https://github.com/odoo/odoo/blob/fc2e80cb4bcc450762c7ac5cb82a3e2d88062b38/addons/web/static/src/less/webclient_layout.less#L54), just nested differently

**Comment by pedrobaeza** [comment]
Uhm, entering as demo user, I don't see any app nor menu...

**Comment by lasley** [comment]
Should demo user have access to `Settings` or `Apps` menus? I installed a few apps and it seems to work fine.

image

**Comment by lasley** [comment]
Taking it a step further, I added `Settings` to `Demo User` Admin access rights & answered my own question. Demo user does not have access to the initial 2 app icons - but can see them once the change is made:

image

**Comment by pedrobaeza** [comment]
But at least "Discuss" app should be available I think...

**Comment by lasley** [comment]
Nope - discuss is an additional install it seems. This is a screencap from a fresh installed DB:

image

**Review by pedrobaeza — APPROVED** [review]
Thanks for checking.

**Review by oscarolar — APPROVED** [review]
Thanks!

**Comment by pedrobaeza** [comment]
@Yajo, please check

**Review by yajo — CHANGES_REQUESTED** [review]
Enter mobile mode as demo user and choose "Sales". When the submenu is hidden, you see this:

captura de pantalla de 2016-10-19 09-18-47

using firefox

**Comment by lasley** [comment]
Well that's sad - I have to fully remove the oe_main_menu_navbar id in order to stop a bunch of the core JS from firing. This resolved the [ @Yajo comment](https://github.com/OCA/web/pull/446#pullrequestreview-4814058)

I don't see an easy way around this & there wasn't a particular reason for me doing it outside of simply wanting the previous ID to exist for other modules that may target it, so I removed & reverted the styling fixes associated with it.

**Comment by pedrobaeza** [comment]
Waiting for @Yajo's confirmation to merge.

**Comment by lasley** [comment]
Once he confirms, I would like to squash the commits logically as well in order to allow for easier cherry-picking into the forward/backports & to not conflict with our OCD for a clean diff 😄

**Review by yajo — APPROVED** [review]
It's sad that you had to remove that, but this now works, so it's OK.

Maybe you could add that to known issues if you want, but it's OK anyway.

Waiting until you rebase to merge.

**Comment by lasley** [comment]
I agree that it's sad I had to remove it - I was pretty excited when I added it, particularly when I noticed the styling differences.

Adding to known issues is a good call, I can see that being trying again later. Done.

Rebase is done as well, turns out the logical commit was simply the one. Was worth the check though IMO - I believe we're good for merge once we get CI 🍏

**Pull request merged**
